### PR TITLE
Form field tel using JLayouts

### PR DESCRIPTION
**Pull request by dgrammatiko**
#### Separate logic/ output

Base work for any serious work for template update etc..
#### Summary of Changes

Introduce a  layout for this field
#### Testing Instructions

Apply patch and rename any backend input to tel
eg

``` XML
<field name="mytextvalue" type="tel" default="Some text" label="Enter some text" description="" size="10" />
```

Also consult: https://docs.joomla.org/Standard_form_field_types

### CODE REVIEW

File: layouts/joomla/form/field/tel.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ * @var   integer  $maxLength       The maximum length that the field shall accept.
+ */
+
+// Including fallback code for HTML5 non supported browsers.
+JHtml::_('jquery.framework');
+JHtml::_('script', 'system/html5fallback.js', false, true);
+
+$autocomplete = !$autocomplete ? ' autocomplete="off"' : ' autocomplete="' . $autocomplete . '"';
+
+$attributes = [
```
Comment (mbabker) [inline layouts/joomla/form/field/tel.php]: Short array syntax

Comment (dgt41) [inline layouts/joomla/form/field/tel.php]: php 5.3  or code style?

Comment (mbabker) [inline layouts/joomla/form/field/tel.php]: Both.  Still supporting PHP 5.3 so can't use short array syntax.

**Comment by laoneo** [comment]
Tests are failing.

**Comment by dgt41** [comment]
@laoneo the msg is: `Undefined property: JLayoutFile::$value` but I can't run phpunit locally without an error `Fatal error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 20480 bytes) in /Users/dimitris/Documents/github_projects/joomla1/libraries/joomla/log/entry.php on line 103`

Any hints are welcome

**Comment by laoneo** [comment]
Glad it is passing now and that github allows to squash pull requests 😅

**Comment by dgt41** [comment]
That looks better 😄
@laoneo you realise that I used your code (from your PR for pagination) for eliminating the spaces in the UT?

**Comment by laoneo** [comment]
Yes saw it. Perhaps we need to find a way to improve that situation. Otherwise we have to copy that all the time. Any suggestions?

**Comment by dgt41** [comment]
maybe use a dom parser?

**Comment by dgt41** [comment]
something like https://github.com/paquettg/php-html-parser

**Comment by mbabker** [comment]
> I can't run phpunit locally without an error Fatal error: Allowed memory size of 134217728 bytes exhausted

IIRC PHP has to have something around 320M memory to run the CMS' test suite in full without the code coverage reports.

**Comment by laoneo** [comment]
@dgt41 good idea, will have a look tomorrow if I can come up with a proper solution.

**Comment by laoneo** [comment]
Idiot me, PHPUnit comes with XML string asserts. Have a look on https://github.com/joomla/joomla-cms/pull/10982/commits/7731ef817b95b080f87c54ea36631dca9716c2b0. There is no need anymore for the ugly str_replace code :triumph:

**Comment by dgt41** [comment]
I will apply this to all the modified tests (damn, there are a lot of them...)

**Comment by brianteeman** [comment]
Maybe strip the tests from the layout PRs and do all the uit tests as one
Pr ?

On 6 July 2016 at 09:43, Dimitri Grammatikogianni <notifications@github.com>
wrote:

> I will apply this to all the modified tests (damn, there are a lot of
> them...)
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/10994#issuecomment-230713372>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABPH8Ss6taMN8OtMiR7DGaWF2B1bixVvks5qS2q-gaJpZM4JDtue>
> .
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by laoneo** [comment]
When we move the hardcoded HTML output to layout's, the tests start failing, that's why the need to be updated within the PR.

**Comment by hardiktailored** [comment]
I applied patched and tested input field with type 'tel'. Nothing new, works same as normal text field. How to check tel validations? Rules?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10994">issues.joomla.org/joomla-cms/10994</a>.</sub>

**Pull request closed without merging**